xbmc/dialogs/GUIDialogMediaFilter.cpp

```diff
@@ -105,6 +105,7 @@ static const CGUIDialogMediaFilter::Filter filterList[] = {
 
   { "albums",       FieldAlbum,         556,    SettingType::String,  "edit",   "string",   CDatabaseQueryRule::OPERATOR_CONTAINS },
 //  { "albums",       FieldArtist,        557,    SettingType::List,    "list",   "string",   CDatabaseQueryRule::OPERATOR_EQUALS },
+  { "albums",       FieldDiscTitle,     38076,  SettingType::List,    "list",   "string",   CDatabaseQueryRule::OPERATOR_EQUALS },
```

Comment:
I put it there as album -> disc seems to make logical sense, but I'm not too bothered about the position of it.  I'm not sure about making it an edit field rather than a pick list though.  Currently it helpfully shows me how many titled discs I have (I was quite surprised to find it's 122 unique titles) and then lets me pick one or more.  Sure, I could type in `Bonus` to achieve the same result, but then I would have to know beforehand what I'm looking for (ie at least one name of a disc) and with a pre-populated list I don't. If it's an editable field, although it's more flexible with 'contains' I think it's easier for a user to miss something that way.  I have found at least 2 albums that I didn't remember had Bonus Discs included with them. A list lets you pick from stuff you already have as album artist does. Also, typing with a remote (amp / tv style remote, not Kore) is tedious and makes me less inclined to use a feature.  I'm aware of the speed impact of it but from a purely personal point of view, that's a tradeoff I'm willing to make.  The average user maybe less so...

Reply:
>Currently it helpfully shows me how many titled discs I have ... and then lets me pick one or more

Then let's have a discs node that can do that without the overheads on the filter dialog.

> I'm aware of the speed impact of it but from a purely personal point of view, that's a tradeoff I'm willing to make. The average user maybe less so...

Yeap, lack of responsiveness  is an issue for others even if that list brings you joy (which I think we can deliver more efficiently another way, so not wanting to deprive you of functionality). Filtering is seriously under used because it is just too slow, I really don't want to make that worse.

Reply:
I hardly ever used filtering before this. To be fair, in spite of it being slow, I have found some stuff I had totally forgotten about whilst testing because of me mixing up stuff in an effort to break what I'd done.

Generally though, I either navigate via recently added or some custom node that I have built or a smart playlist.  Sometimes it's artist -> album but it's rare I'd filter for a specific genre or or year or something.  I'd use a smart playlist to do that.   Hmmm, I really don't want to push for something that makes the user experience worse not better.
I'll switch it to an edit field and see what it's like in use.

Reply:
Yes do try it. If it helps I'll push a new branch and you can just grab that and test.

Also think that the filter dialog is a weak area I intend to make stronger. Larger libraries need a fast flexible filter route into them that are a joy to explore with (so you can find the lost and see your collection in a wide variety of ways). The current implementation doesn't provide that, but I don't want to make even worse in the meantime.

Reply:
I'm rather surprised but I prefer it as an edit field !!  Didn't think I would as I was pretty convinced that a list was the way to go, but I think I've just convinced myself that this way is better.